src/featbin/extract-feature-segments.cc

```diff
@@ -133,10 +149,13 @@ int main(int argc, char *argv[]) {
       const Matrix<BaseFloat> &feats = feat_reader.Value(utterance);
       int32 num_samp = feats.NumRows(), // total number of samples present in wav data
           num_chan = feats.NumCols(); // total number of channels present in wav file
-
       // Convert start & end times of the segment to corresponding sample number
-      int32 start_samp = static_cast<int32>(start * samp_freq);
-      int32 end_samp = static_cast<int32>(end * samp_freq);
+      int32 start_samp = static_cast<int32>(round((start * samp_freq)));
```

Comment:
Also, please co-ordinate carefully with the NumFrames() function.  If you can actually make use of that function directly, it would be ideal, as it would be ideal-- you could create a temporary object of type FrameExtractionOptions inside the code.  (you could just take the sampling rate to be 16k as it wouldn't affect the results if you assume an exact number of samples go into the frame shift and frame length).